simulariumio/trajectory_converter.py

```diff
@@ -81,6 +81,40 @@ def check_report_progress(self, percent_complete: float) -> None:
             self.progress_callback(percent_complete)
             self.last_report_time = current_time
 
+    @staticmethod
+    def check_max_min_coordinates(
+        max_dimensions: np.array,
+        min_dimensions: np.array,
+        current_position: np.array,
+        radius: float = 0.0,
+    ):
+        for i in range(len(current_position)):
+            curr_val = float(current_position[i])
+            if curr_val - radius < min_dimensions[i]:
+                min_dimensions[i] = curr_val - radius
+            if curr_val + radius > max_dimensions[i]:
+                max_dimensions[i] = curr_val + radius
+
+    @staticmethod
+    def calculate_scale_factor(
+        max_dimensions: np.array,
+        min_dimensions: np.array,
+    ) -> float:
+        """
+        Return a scale factor, using the given min and max
+        dimensions, so that the final range of agent locations
+        is within the dimensions defined by VIEWER_DIMENSION_RANGE
+        """
+        range = max(max_dimensions - min_dimensions)
+        scale_factor = 1
+        if range == 0:
```

Comment:
It makes sense to check this so you don't divide by zero, but since it's a float it's unlikely to be exactly zero, so this line seems weird to me. You could use `numpy.isclose()` but also maybe it doesn't matter since division will work fine if it's slightly off zero.

Reply:
I would agree with picking an epsilon here!